src/k8s-engine/wsl.ts

```diff
@@ -985,7 +989,7 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
             const integrations = await this.listIntegrations();
 
             for (const [distro, status] of Object.entries(integrations)) {
-              if (status === true) {
+              if (config.containerEngine === ContainerEngine.MOBY && status === true) {
```

Comment:
I don't get this... if we're running `containerd` we always want to stop the integrated process?  If you don't want to run integrations for containerd then why not wrap the whole `progressTracker.action` block in a check for `MOBY`? 

Looks like `integrationProcesses` should be `dockerProxyProcesses` because it isn't used for anything else.  They aren't used in the wsl implementation of `listIntegrations` so I don't see the harm in renaming it.

And why isn't `this.mobySocketProxyProcess = new BackgroundProcess(...)` guarded similarly?

Reply:
For the last part: we can _create_ background processes, because until we actually call `start()` on it, they don't actually do anything. We do guard the `start()` call on the container engine.

But you're right, I'll fix it so that it's all behind the same check.

Reply:
While trying to fix this… this one is guarded because the user may turn _off_ integrations (after having previously turned it on).  Or, indeed, turn integrations _on_ in the middle of startup.